[structure]: Make basic structure

Derek-94 opened the pull request:
### Details

- Add ProductHistory component class structure.
  - renders differently if there is data in localstorage.
- Add Routers.

### 구현내용

![image](https://user-images.githubusercontent.com/52649378/127612015-e8981b55-52f7-4299-8eff-ad398d534cce.png)

이렇게 라우터를 추가했습니다! 편하신 네이밍으로 개발 시작하셔도될것같아요.

@jamieecode 재완님께서 이미 시작하신것 같은데, 이미 코드를 추가하셨으면 커밋 남기고 `git pull origin develop` 하시고 개발하시면 됩니다.

closes #5

Derek-94 committed:
feat(productHistory): Update basic structure of productHistory class.

- Update class component.
- Renders differently in terms of localStorage status.

Derek-94 committed:
Merge branch 'develop' of https://github.com/Derek-94/mr.camel into feature/ProductHistory#5

Derek-94 committed:
feat(router): Add Router module.

blingblin-g commented on src/index.js:
```diff
@@ -1,10 +1,13 @@
-    <Router />
-  </React.StrictMode>,
+  <BrowserRouter>
+    <React.StrictMode>
```
왜 BroswerRouter가 StrictMode 상위에 있나요?

Derek-94 replied:
아 실수했습니다. 죄송합니다. 시정하겠습니다.

blingblin-g commented on src/Components/ProductHistory.jsx:
```diff
@@ -0,0 +1,100 @@
+export default class ProductHistory extends Component {
+  constructor(props) {
+    super();
+    localStorage.setItem('productHistory', JSON.stringify(tmp));
```
local storage 관련 함수 제가 utils에 따로 파서 만들겠습니다.

Derek-94 replied:
넵 감사합니다. 키 값 관련, 그리고 어떻게 생긴 객체인지는 쓰레드로 의견 공유 부탁드릴게요.

6번째줄은 저희가 테스트 용도로 적었습니다. 개발시에만 적용하고, 지워질 코드입니다.

blingblin-g commented on src/Router.js:
```diff
@@ -1,14 +1,27 @@
-        {/* <Route exact path="/" component={} />
-        <Route path="/product" component={} />
-        <Route path="/recentList" component={} /> */}
+        <ul>
```
네비게이션 기능과 라우터 기능이 같이 있는 것 같은데..
하나의 컴포넌트에 하나의 기능만 있어야할 것 같습니다

Derek-94 replied:
큰 틀만 정해둔거라서요. 그럼 `Router`에는 `<Route>` 만 있는 방향으로 수정하겠습니다.

blingblin-g reviewed (commented):
댓글 남겼습니다.

Derek-94 reviewed (commented):
답변달았습니다. 코드 수정해서 다시 푸쉬할게요.

Derek-94 committed:
fix: Applied feedbacks.

- Change order of BrowseRouter, strickMode.
- Substract List of Link out of Router.
